src/java-interop/java-interop-dlfcn.cc

```diff
@@ -0,0 +1,173 @@
+#include "java-interop.h"
+#include "java-interop-dlfcn.h"
+#include "java-interop-util.h"
+
+#ifdef WINDOWS
+#include <libloaderapi.h>
+#include <winerror.h>
+#include <wtypes.h>
+#include <winbase.h>
+#else
+#include <dlfcn.h>
+#endif
+
+static char *
+_get_last_dlerror ()
+{
+#ifdef WINDOWS
+
+	DWORD error = GetLastError ();
+	if (error == ERROR_SUCCESS /* 0 */) {
+		return nullptr;
+	}
+
+	wchar_t *buf = nullptr;
+
+	DWORD size = FormatMessageW (
+			/* dwFlags */       FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM |FORMAT_MESSAGE_IGNORE_INSERTS,
+			/* lpSource */      NULL,
+			/* dwMessageId */   error,
+			/* dwLanguageId */  MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+			/* lpBuffer */      (LPWSTR) &buf,
+			/* nSize */         0,
+			/* Arguments */     NULL
+	);
+	if (size == 0)
+		return nullptr;
+
+	char *message = utf16_to_utf8 (buf);
+	LocalFree (buf);
+
+	return message;
+
+#else   // ndef WINDOWS
+
+	return java_interop_strdup (dlerror ());
+
+#endif  // ndef WINDOWS
+}
+
+void*
+java_interop_load_library (const char *path, unsigned int flags, char **error)
+{
+	if (error) {
```

Comment:
for consistency with line 57 (and other `path` `nullptr` checks,) all the `if (error)` statements should use `if (error != nullptr)` in this function and the ones below, IMO

Reply:
This has been done.